tests/Find-DbaDbUnusedIndex.Tests.ps1

```diff
@@ -1,19 +1,136 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+$script:UnusedIndexCommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
 Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
 . "$PSScriptRoot\constants.ps1"
 
-Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
+Describe "$script:UnusedIndexCommandName Unit Tests" -Tag 'UnitTests' {
     Context "Validate parameters" {
-        [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object {$_ -notin ('whatif', 'confirm')}
-        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Database', 'ExcludeDatabase', 'IgnoreUptime', 'InputObject', 'EnableException'
-        $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
         It "Should only contain our specific parameters" {
-            (@(Compare-Object -ReferenceObject ($knownParameters | Where-Object {$_}) -DifferenceObject $params).Count ) | Should Be 0
+            [object[]]$params = (Get-Command $script:UnusedIndexCommandName).Parameters.Keys | Where-Object { $_ -notin ('whatif', 'confirm') }
+            [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Database', 'ExcludeDatabase', 'IgnoreUptime', 'InputObject', 'EnableException', 'UserSeeksLessThan', 'UserScansLessThan', 'UserLookupsLessThan'
+            $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
+
+            (@(Compare-Object -ReferenceObject ($knownParameters | Where-Object { $_ }) -DifferenceObject $params).Count ) | Should -Be 0
         }
     }
 }
 <#
     Integration test should appear below and are custom to the command you are writing.
     Read https://github.com/sqlcollaborative/dbatools/blob/development/contributing.md#tests
     for more guidence.
-#>
\ No newline at end of file
+#>
+
+
+Describe "$script:UnusedIndexCommandName Integration Tests" -Tags "IntegrationTests" {
+    Context "Verify basics of the Find-DbaDbUnusedIndex command" {
+        BeforeAll {
+            $server1 = Connect-DbaInstance -SqlInstance $script:instance1
+            $server2 = Connect-DbaInstance -SqlInstance $script:instance2
+            $server3 = Connect-DbaInstance -SqlInstance $script:instance3
```

Comment:
Please only perform the test on instance3.

Reply:
Sure, np

Reply:
There seems to be an issue with using instance3. The [default scenario](https://ci.appveyor.com/project/sqlcollaborative/dbatools/builds/34725387/job/in72ya3mndc7e8p9/artifacts) in AppVeyor is getting selected and then at runtime there is an instance failure when trying to connect to the sql2017 instance. Is there an issue with the dynamic scenario selection? If I use [instance2](https://ci.appveyor.com/project/sqlcollaborative/dbatools/builds/34726052/job/4ebaf7299tdf8uud) the test works fine.

Reply:
Your last run shows the test running under the excepted scenario for sql2016_sql2017